src/data-workspace/data-details-sidebar/comment.js

```diff
@@ -83,17 +93,44 @@ const CommentEditField = ({ comment, commentOptionSetId }) => {
 
 CommentEditField.propTypes = {
     comment: PropTypes.string,
+    commentId: PropTypes.string,
     commentOptionSetId: PropTypes.string,
+    unsavedComment: PropTypes.string,
+    onBlur: PropTypes.func,
 }
 
-function CommentEditForm({ item, open, setOpen, syncComment, closeEditor }) {
+function CommentEditForm({
+    item,
+    open,
+    setOpen,
+    syncComment,
+    closeEditor,
+    unsavedComment,
+    commentId,
+}) {
+    const contextSelectionId = useContextSelectionId()
+    const cellId = getCellId({ contextSelectionId, item })
+    const setUnsavedComment = useUnsavedDataStore(
+        (state) => state.setUnsavedComment
+    )
+
     const onSubmit = (values) => {
         // Don't send `undefined` (or 'undefined' will be stored as the comment)
-        const comment = values.comment || ''
+        const comment = values[`comment_${commentId}`] || ''
         syncComment({ comment })
```

Comment:
We're not really guaranteed this succeeds right?
Due to the shared-data-value-mutation there needs some refactor to pass an `onSuccess`. But we could use the `mutateAsync`-instead. Se comment below

Reply:
I, see thanks for that!

While testing I noticed that the entire form is replaced by a spinner when saving. I know that this was not introduced in this PR, but it's an unfortunate cumulative layout shift. I think we should move this "saving"-logic down to the form, and show a spinner on the button instead.